frameworks/elastic/tests/test_kibana.py

```diff
@@ -1,47 +1,150 @@
-        assert rc == 0 and decoded_base_64_yml in stdout
-
-        config.check_kibana_adminrouter_integration(
-            "service/{}/".format(config.KIBANA_SERVICE_NAME)
-        )
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+        sdk_install.uninstall(config.KIBANA_PACKAGE_NAME, config.KIBANA_SERVICE_NAME)
+
+        yield  # let the test session execute
     finally:
-        _uninstall_services()
+        sdk_install.uninstall(config.PACKAGE_NAME, config.SERVICE_NAME)
+        sdk_install.uninstall(config.KIBANA_PACKAGE_NAME, config.KIBANA_SERVICE_NAME)
+
+
+@pytest.mark.sanity
+def test_config_with_custom_yml(configure_package) -> None:
+    sdk_install.install(
+        config.PACKAGE_NAME,
+        service_name=config.SERVICE_NAME,
+        expected_running_tasks=config.DEFAULT_TASK_COUNT,
+    )
+
+    decoded_base_64_yml = "logging.json: true"
+    base_64_yml = base64.b64encode(decoded_base_64_yml.encode("utf-8")).decode("utf-8")
+    elasticsearch_url = "http://" + sdk_hosts.vip_host(config.SERVICE_NAME, "coordinator", 9200)
+
+    sdk_install.install(
+        config.KIBANA_PACKAGE_NAME,
+        config.KIBANA_SERVICE_NAME,
+        0,
+        {"kibana": {"elasticsearch_url": elasticsearch_url, "custom_kibana_yml": base_64_yml}},
+        wait_for_deployment=False,
+        insert_strict_options=False,
+    )
+
+    cmd = "bash -c 'grep \"{}\" kibana-*/config/kibana.yml'".format(decoded_base_64_yml)
+    rc, stdout, stderr = sdk_cmd.marathon_task_exec(config.KIBANA_SERVICE_NAME, cmd)
+    assert rc == 0 and decoded_base_64_yml in stdout
+
+    config.check_kibana_adminrouter_integration("service/{}/".format(config.KIBANA_SERVICE_NAME))
+
+
+@pytest.mark.sanity
+def test_virtual_network(configure_package) -> None:
+    sdk_install.install(
+        config.PACKAGE_NAME,
+        service_name=config.SERVICE_NAME,
+        expected_running_tasks=config.DEFAULT_TASK_COUNT,
+        additional_options=sdk_networks.ENABLE_VIRTUAL_NETWORKS_OPTIONS,
+    )
+
+    elasticsearch_url = "http://" + sdk_hosts.vip_host(config.SERVICE_NAME, "coordinator", 9200)
+    sdk_install.install(
+        config.KIBANA_PACKAGE_NAME,
+        config.KIBANA_SERVICE_NAME,
+        0,
+        {
+            "service": {
+                "virtual_network_enabled": True,
+                "virtual_network_plugin_labels": [
+                    {"key": "c", "value": "d"},
+                    {"key": "a", "value": "b"},
+                ],
+            },
+            "kibana": {"elasticsearch_url": elasticsearch_url},
+        },
+        wait_for_deployment=False,
+        insert_strict_options=False,
+    )
+
+    _check_cni_working()
+
+
+def _check_cni_working() -> None:
+    config.check_kibana_adminrouter_integration("service/{}/".format(config.KIBANA_SERVICE_NAME))
+
+    kibana_config = sdk_marathon.get_config(config.KIBANA_SERVICE_NAME)
+    expected_labels = {"c": "d", "a": "b", "network_name": "dcos"}
```

Comment:
Instead of hard-coding the same label set in two different places (tests and `_check_cni_working()`) would it make sense to set them in the test and pass them as a parameter to `_check_cni_working()`?

Reply:
Refactored. Thanks!